pypeit/slittrace.py

```diff
@@ -1401,93 +1418,102 @@ def get_maskdef_objpos_offset_alldets(sobjs, calib_slits, spat_flexure, platesca
-                if calib_dets[i] in red_dets:
-                    red_slitmask_offsets.append(slitmask_offsets[i])
-            blue_slitmask_offsets = np.array(blue_slitmask_offsets)
-            red_slitmask_offsets = np.array(red_slitmask_offsets)
-
-            # compute average slitmask_offset for blue detectors
-            # msgs.warn('Slitmask offsets in each blue det: {}.'.format(np.round(blue_slitmask_offsets, 2)))
-            if blue_slitmask_offsets[blue_slitmask_offsets!=0].size > 0:
-                _, median_off_blue, _ = sigma_clipped_stats(blue_slitmask_offsets[blue_slitmask_offsets!=0], sigma=2.)
-            # compute average slitmask_offset for red detectors
-            # msgs.warn('Slitmask offsets in each red det: {}.'.format(np.round(red_slitmask_offsets, 2)))
-            if red_slitmask_offsets[red_slitmask_offsets!=0].size > 0:
-                _, median_off_red, _ = sigma_clipped_stats(red_slitmask_offsets[red_slitmask_offsets!=0], sigma=2.)
-
-            # if all the blue_slitmask_offsets were zero, we estimate median_off_blue from median_off_red
-            # usually traces in red dets are 1-2pixels shifted to the right w.r.t. traces in blue dets
-            if 'median_off_blue' not in locals():
-                median_off_blue = median_off_red - 1.
-            # if all the red_slitmask_offsets were zero, we estimate median_off_red from median_off_blue
-            if 'median_off_red' not in locals():
-                median_off_red = median_off_blue + 1.
-            # at this point, we should never have both blue_slitmask_offsets and red_slitmask_offsets all zero.
-
-            if np.any([det in blue_dets for det in calib_dets]):
-                msgs.info('Average Slitmask offset for the blue detectors: '
-                          '{:.2f} pixels ({:.2f} arcsec).'.format(median_off_blue, median_off_blue*platescale))
-            if np.any([det in red_dets for det in calib_dets]):
-                msgs.info('Average Slitmask offset for the red detectors: '
-                          '{:.2f} pixels ({:.2f} arcsec).'.format(median_off_red, median_off_red*platescale))
-
-            # update the slit calibration with the average slitmask offset
-            for i in range(len(calib_slits)):
-                if calib_slits[i].det in blue_dets:
-                    calib_slits[i].maskdef_offset = median_off_blue
-                if calib_slits[i].det in red_dets:
-                    calib_slits[i].maskdef_offset = median_off_red
+            # if median was not computed for blue dets, use the remaining (red) dets to compute the median for all dets
+            _, median_off, _ = sigma_clipped_stats(slitmask_offsets, sigma=2.)
+            for cs in calib_slits:
+                cs.maskdef_offset = median_off
+            msgs.info('Average Slitmask offset for all the detectors: '
+                      '{:.2f} pixels ({:.2f} arcsec).'.format(median_off, median_off * platescale))
```

Comment:
I think you can combine this block and the identical block at line 1474.  I think starting with line 1472, you could do:
```
    indx_b = np.where(np.in1d(calib_dets, spectrograph_dets[0]))[0]
    if not blue_and_red or slitmask_offsets[indx_b].size == 0:
        _, median_off, _ = sigma_clipped_stats(slitmask_offsets, sigma=2.)
        for cs in calib_slits:
            # assign median to each det
            cs.maskdef_offset = median_off
        msgs.info('Average Slitmask offset: {:.2f} pixels ({:.2f} arcsec).'.format(median_off, median_off * platescale))
        return calib_slits

    # compute median if these dets have values of slitmask_offsets
    _, median_off, _ = sigma_clipped_stats(slitmask_offsets[indx_b], sigma=2.)
    for cs in calib_slits[indx_b]:
        ...
```

Reply:
Done. Thank you!